### PR TITLE
Make some errors serializable

### PR DESCRIPTION
**Pull request by sagudev**
**Description**
In servo we need errors to be serializable.

Reviewable per commits (each commit builds by itself)

**Testing**
It compiles.

**Checklist**

- [x] Run `cargo fmt`.
- [x] Run `cargo clippy`. If applicable, add:
  - [ ] `--target wasm32-unknown-unknown`
  - [ ] `--target wasm32-unknown-emscripten`
- [x] Run `cargo xtask test` to run tests.
- [ ] Add change to `CHANGELOG.md`. See simple instructions inside file.

### CODE REVIEW

File: wgpu-core/src/instance.rs [hunk 1]
```diff
@@ -391,6 +392,8 @@ pub enum GetSurfaceSupportError {
 }
 
 #[derive(Clone, Debug, Error)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "serde", serde(bound(deserialize = "'de: 'static")))]
```
Comment (ErichDonGubler) [inline wgpu-core/src/instance.rs:396]: question: These `bound` attributes seem unnecessary. Does [`serde::de::DeserializeOwned`](https://docs.rs/serde/1.0.204/serde/de/trait.DeserializeOwned.html) not resolve this problem?

**Comment by ErichDonGubler** [comment]
I don't see any blocking issues with this. I want to tag in @cwfitzgerald for thoughts on supporting this as a temperature check, but I don't see any issues with the implementation beyond what I've submitted comments for.

**Review by ErichDonGubler — APPROVED** [review]

**Comment by cwfitzgerald** [comment]
No problems with this landing, the only thing I'm unsure about is that these serializes have no in-codebase justification, so could be broken at some point. I don't know what to do about that problem, just noting it.

File: wgpu-core/src/instance.rs (hunk 1 above)
Comment (sagudev) [reply wgpu-core/src/instance.rs:396]: There is no way to derive `DeserializeOwned` (I think it only serves as a bound for consumers not implementers) or am I missing something?

**Comment by sagudev** [comment]
> No problems with this landing, the only thing I'm unsure about is that these serializes have no in-codebase justification, so could be broken at some point. I don't know what to do about that problem, just noting it.

We could add test (or const asserts) that some types implements (de)serialize (like it's done for Sync/Send), but I think that would make things complicated. (de)serialize can be trivially added again if needed by consumers (mostly browsers).

**Review by cwfitzgerald — APPROVED** [review]
Lezdoit

**Pull request merged**